testsuite/MDAnalysisTests/utils/test_log.py

```diff
@@ -81,106 +63,113 @@ def __exit__(self, exc_type, exc_value, traceback):
-            for frame in range(n):
-                rmsd = 0.02 * frame * (n+1)/float(n)  # n+1/n correction for 0-based frame vs 1-based counting
-                pm.echo(frame, rmsd=rmsd)
-        self.buf.seek(0)
-        output = "".join(self.buf.readlines())
-        self._assert_in(output,
-                        ('\r' + format) % {'rmsd': 0.0, 'step': 1,
-                                           'numsteps': n, 'percentage': 100./n})
-        # last line always ends with \n!
-        self._assert_in(output,
-                        ('\r' + format + '\n') % {'rmsd': 0.02*n, 'step': n,
-                                                  'numsteps': n, 'percentage': 100.0})
-
-    def test_not_dynamic_ProgressMeter(self, n=51, interval=10):
-        format = "Step {step:5d}/{numsteps} [{percentage:5.1f}%]"
-        with RedirectedStderr(self.buf):
-            pm = MDAnalysis.lib.log.ProgressMeter(n, interval=interval,
-                                                  dynamic=False)
-            for frame in range(n):
-                pm.echo(frame)
-        self.buf.seek(0)
-        output = "".join(self.buf.readlines())
-        self._assert_in(output, (format + '\n').format(**{'step': 1, 'numsteps': n, 'percentage': 100./n}))
-        self._assert_in(output, (format + '\n').format(**{'step': n, 'numsteps': n, 'percentage': 100.}))
-
-
-def test__set_verbose():
-    # Everything agrees verbose should be True
-    assert_equal(_set_verbose(verbose=True, quiet=False, default=True), True)
-    # Everything agrees verbose should be False
-    assert_equal(_set_verbose(verbose=False, quiet=True, default=False), False)
-    # Make sure the default does not overwrite the user choice
-    assert_equal(_set_verbose(verbose=True, quiet=False, default=False), True)
-    assert_equal(_set_verbose(verbose=False, quiet=True, default=True), False)
-    # Quiet is not provided
-    assert_equal(_set_verbose(verbose=True, quiet=None, default=False), True)
-    assert_equal(_set_verbose(verbose=False, quiet=None, default=False), False)
-    # Verbose is not provided
-    assert_equal(_set_verbose(verbose=None, quiet=True, default=False), False)
-    assert_equal(_set_verbose(verbose=None, quiet=False, default=False), True)
-    # Nothing is provided
-    assert_equal(_set_verbose(verbose=None, quiet=None, default=True), True)
-    assert_equal(_set_verbose(verbose=None, quiet=None, default=False), False)
-    # quiet and verbose contradict each other
-    assert_raises(ValueError, _set_verbose, verbose=True, quiet=True)
-    assert_raises(ValueError, _set_verbose, verbose=False, quiet=False)
-    # A deprecation warning is issued when quiet is set
 
+@pytest.fixture()
+def buffer():
+    return StringIO()
+
+
+def _assert_in(output, string):
+    assert string in output, "Output '{0}' does not match required format '{1}'.".format(output.replace('\r', '\\r'), string.replace('\r', '\\r'))
+
+
+def test_default_ProgressMeter(buffer, n=101, interval=10):
```

Comment:
The `test_*_ProgressMeter` functions are basicaly all doing the same thing. They should be parametrized.